File: app/src/main/java/com/bq/zowi/models/commands/StopCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class StopCommand extends MotionlessCommand {
    private static final String STOP = "S";
    private static final Command.Action[] STOP_ACTIONS = {Command.Action.STOP};

    public StopCommand() {
        setAction(Command.Action.STOP);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return STOP_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public String getCommandValue() {
        return "S\r\n";
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new StopCommand();
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/StopCommand.kt
package com.bq.zowi.models.commands

class StopCommand : MotionlessCommand() {

    init {
        setAction(Command.Action.STOP)
    }

    override fun getAllowedActions(): Array<Command.Action> = STOP_ACTIONS
    override fun getCommandValue(): String = "S\r\n"
    override fun copy(): Command = StopCommand()

    companion object {
        private val STOP_ACTIONS = arrayOf(Command.Action.STOP)
    }
}
